cmd/runner/main.go

```diff
@@ -53,16 +70,294 @@ func run(args []string) error {
 	if err != nil {
 		return fmt.Errorf("failed to init logger: %w", err)
 	}
+	defer logger.Sync()
+
+	var creds azcore.TokenCredential
+	// Azure has 3-4 ways to authenticate as an identity to their APIs (KMS, storage, etc).
+	//   - When running locally, we use the "Environment" approach, which means we provide AZURE_* environment variables that authenticate against a local-only service account.
+	//   - When running in Azure Container Apps Jobs, we use the "ManagedIdentitiy" approach, meaning we pull ambiently from the infrastructure we're running on (via a metadata service).
+	// See https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#readme-credential-types for more info
+	if azClientSecret := os.Getenv("AZURE_CLIENT_SECRET"); azClientSecret != "" {
+		if creds, err = azidentity.NewEnvironmentCredential(nil); err != nil {
+			return fmt.Errorf("failed to load Azure credentials from environment: %w", err)
+		}
+	} else {
+		// We use "ManagedIdentity" instead of just "Default" because the default
+		// timeout is too low in azidentity.NewDefaultAzureCredentials, so it times out
+		// and fails to run.
+		azClientID := os.Getenv("AZURE_CLIENT_ID")
+		logger.Info("Loading user managed credentials", zap.String("client_id", azClientID))
+		if creds, err = azidentity.NewManagedIdentityCredential(&azidentity.ManagedIdentityCredentialOptions{
+			ID: azidentity.ClientID(azClientID),
+		}); err != nil {
+			return fmt.Errorf("failed to load Azure credentials: %w", err)
+		}
+	}
+
+	blobClient, err := azblob.NewClient(creds, *azStorageAccount)
+	if err != nil {
+		return fmt.Errorf("failed to init blob client: %w", err)
+	}
+
+	h := handler{
+		blob:                     blobClient,
+		sourcePortfolioContainer: *azSourcePortfolioContainer,
+		destPortfolioContainer:   *azDestPortfolioContainer,
+		reportContainer:          *azReportContainer,
+	}
+
+	validTasks := map[task.Type]func(context.Context) error{
+		task.ProcessPortfolio: toRunFn(processPortfolioReq, h.processPortfolio),
+		task.CreateReport:     toRunFn(createReportReq, h.createReport),
+	}
+
+	taskType := task.Type(os.Getenv("TASK_TYPE"))
+	if taskType == "" {
+		return errors.New("no TASK_TYPE given")
+	}
+
+	taskFn, ok := validTasks[taskType]
+	if !ok {
+		return fmt.Errorf("unknown task type %q", taskType)
+	}
+
+	logger.Info("running PACTA task", zap.String("task_type", string(taskType)))
+
+	if err := taskFn(ctx); err != nil {
+		return fmt.Errorf("error running task: %w", err)
+	}
+
+	logger.Info("ran PACTA task successfully", zap.String("task_type", string(taskType)))
```

Comment:
Optional: maybe include some structured logging info here about runtime?

Reply:
Added the `task_id` so we can correlate the timestamps on the log more easily